flink-table/flink-table-planner/src/test/java/org/apache/flink/table/planner/plan/nodes/exec/serde/RelDataTypeJsonSerdeTest.java

```diff
@@ -46,83 +35,113 @@
 
-/** Tests for serialization/deserialization of {@link RelDataType}. */
-@RunWith(Parameterized.class)
+/** Tests for {@link RelDataType} serialization and deserialization. */
 public class RelDataTypeJsonSerdeTest {
+
     private static final FlinkTypeFactory FACTORY = FlinkTypeFactory.INSTANCE();
 
-    @Parameterized.Parameters(name = "type = {0}")
-    public static Collection<RelDataType> parameters() {
+    @ParameterizedTest
+    @MethodSource("testRelDataTypeSerde")
+    public void testRelDataTypeSerde(RelDataType relDataType) throws IOException {
+        final SerdeContext serdeContext = configuredSerdeContext();
+
+        final String json = toJson(serdeContext, relDataType);
+        final RelDataType actual = toObject(serdeContext, json, RelDataType.class);
+
+        assertThat(actual).isSameAs(relDataType);
+    }
+
+    @Test
+    public void testMissingPrecisionAndScale() {
+        final SerdeContext serdeContext = configuredSerdeContext();
+
+        final String json =
+                toJson(
+                        serdeContext,
+                        FACTORY.createSqlIntervalType(
+                                new SqlIntervalQualifier(
+                                        TimeUnit.DAY, TimeUnit.SECOND, SqlParserPos.ZERO)));
+        final RelDataType actual = toObject(serdeContext, json, RelDataType.class);
+
+        assertThat(actual)
+                .isSameAs(
+                        FACTORY.createSqlIntervalType(
+                                new SqlIntervalQualifier(
+                                        TimeUnit.DAY,
+                                        DayTimeIntervalType.DEFAULT_DAY_PRECISION,
+                                        TimeUnit.SECOND,
+                                        DayTimeIntervalType.DEFAULT_FRACTIONAL_PRECISION,
+                                        SqlParserPos.ZERO)));
+    }
+
+    // --------------------------------------------------------------------------------------------
+    // Test data
+    // --------------------------------------------------------------------------------------------
+
+    @Parameters(name = "{0}")
+    public static List<RelDataType> testRelDataTypeSerde() {
         // the values in the list do not care about nullable.
-        List<RelDataType> types =
+        final List<RelDataType> types =
                 Arrays.asList(
                         FACTORY.createSqlType(SqlTypeName.BOOLEAN),
                         FACTORY.createSqlType(SqlTypeName.TINYINT),
                         FACTORY.createSqlType(SqlTypeName.SMALLINT),
                         FACTORY.createSqlType(SqlTypeName.INTEGER),
                         FACTORY.createSqlType(SqlTypeName.BIGINT),
-                        FACTORY.createSqlType(SqlTypeName.DECIMAL, 3, 10),
-                        FACTORY.createSqlType(SqlTypeName.DECIMAL, 0, 19),
-                        FACTORY.createSqlType(SqlTypeName.DECIMAL, -1, 19),
```

Comment:
Shouldn't we test the negative value?

Reply:
correct me if I'm wrong, but why should we support negative precision or scale that is smaller than precision? it looks like a bug in the tests to me.

Reply:
I think so, that's why I asked here: https://github.com/apache/flink/pull/18342#discussion_r783771529

Reply:
This covers negative scale. Negative scale we can support. But what should negative precision mean?

Reply:
I can't think of such a case, doesn't make sense.